packages/exchange/src/exchange/providers/nvidia.py

```diff
@@ -0,0 +1,104 @@
+import os
+from typing import Any, Dict, List, Tuple, Type
+
+import httpx
+from httpx._auth import FunctionAuth 
+
+from exchange.message import Message
+from exchange.providers.base import Provider, Usage
+from exchange.providers.utils import (
+    messages_to_openai_spec,
+    openai_response_to_message,
+    openai_single_message_context_length_exceeded,
+    raise_for_status,
+    tools_to_openai_spec,
+)
+from exchange.tool import Tool
+from tenacity import retry, wait_fixed, stop_after_attempt
+from exchange.providers.utils import retry_if_status
+
+NVIDIA_HOST = "https://integrate.api.nvidia.com/"
+
+retry_procedure = retry(
+    wait=wait_fixed(2),
+    stop=stop_after_attempt(2),
+    retry=retry_if_status(codes=[429], above=500),
+    reraise=True,
+)
+
+class NVIDIAProvider(Provider):
+    """Provides chat completions for models exposed through NVIDIA Inference Microservice (NIM) """
+
+    def __init__(self, client: httpx.Client) -> None:
+        super().__init__()
+        self.client = client
+
+    @classmethod
+    def from_env(cls: Type["NVIDIAProvider"]) -> "NVIDIAProvider":
```

Comment:
this works but @elenazherdeva introduced `cls.check_env_vars()` pattern so these can be checked upfront avoiding the need for a `try/except` block 

https://github.com/block-open-source/goose/blob/8276e9b01f1f9ffe17b9388861b52ed96066c3c4/packages/exchange/src/exchange/providers/google.py#L25-L27